### DISCUSSION TITLE
Behavior when changing preset during window is open - the clarification

### DISCUSSION THREAD

**Discussion by yannc74** [jmcollin78/versatile_thermostat#612] [Q&A] [answered]
Hi, 

I don't have the same behavior as last year when I change the preset mode and the window is open.

Scenario:
Initial state:
heating: eco (current temp > target temp)
window open
so, the heating is off

I select "boost" :
heating: boost (current temp < target temp)
window open
so, the heating is on !

**why ? The last year, the heating was always off in this case.**

Initial state:
hvac_modes: heat, off
min_temp: 7
max_temp: 25
target_temp_step: 0.1
preset_modes: none, frost, eco, comfort, boost
current_temperature: 18.4
**temperature: 13
hvac_action: idle
preset_mode: eco**
is_on: true
hvac_mode: heat
type: null
is_controlled_by_central_mode: true
last_central_mode: Stopped
frost_temp: 7
eco_temp: 13
boost_temp: 22
comfort_temp: 20
frost_away_temp: 7
eco_away_temp: 13
boost_away_temp: 20
comfort_away_temp: 18
power_temp: null
target_temperature_step: 0.1
ext_current_temperature: 11
ac_mode: false
current_power: null
current_power_max: null
**saved_preset_mode: frost**
saved_target_temp: 7
saved_hvac_mode: heat
motion_sensor_entity_id: null
motion_state: null
power_sensor_entity_id: null
max_power_sensor_entity_id: null
overpowering_state: null
presence_sensor_entity_id: binary_sensor.presence_domicile
presence_state: off
window_state: on
window_auto_state: off
window_bypass_state: false
window_sensor_entity_id: binary_sensor.fenetre_chambre_ouverture
window_delay_sec: 30
window_auto_enabled: false
window_auto_open_threshold: 3
window_auto_close_threshold: 0
window_auto_max_duration: 30
window_action: window_turn_off
security_delay_min: 60
security_min_on_percent: 0.5
security_default_on_percent: 0.1
last_temperature_datetime: 2024-11-07T11:29:57.718982+01:00
last_ext_temperature_datetime: 2024-11-07T11:29:57.718982+01:00
security_state: false
minimal_activation_delay_sec: 10
device_power: 1000
**mean_cycle_power: 0**
total_energy: 256.41
last_update_datetime: 2024-11-07T11:29:57.719164+01:00
timezone: Europe/Paris
temperature_unit: °C
**is_device_active: false**
ema_temp: 18.4
is_used_by_central_boiler: false
temperature_slope: 0
hvac_off_reason: null
is_over_switch: true
is_inversed: false
keep_alive_sec: 0
underlying_entities: switch.fil_pilote_qubino_chambre_fil_pilote_qubino_chambre_switch
**on_percent: 0
power_percent: 0
on_time_sec: 0
off_time_sec: 300**
cycle_min: 5
function: tpi
tpi_coef_int: 0.6
tpi_coef_ext: 0.01
friendly_name: Thermostat Chambre
supported_features: 401

After:
hvac_modes: heat, off
min_temp: 7
max_temp: 25
target_temp_step: 0.1
preset_modes: none, frost, eco, comfort, boost
current_temperature: 18.4
**temperature: 20
hvac_action: heating
preset_mode: boost**
is_on: true
hvac_mode: heat
type: null
is_controlled_by_central_mode: true
last_central_mode: Stopped
frost_temp: 7
eco_temp: 13
boost_temp: 22
comfort_temp: 20
frost_away_temp: 7
eco_away_temp: 13
boost_away_temp: 20
comfort_away_temp: 18
power_temp: null
target_temperature_step: 0.1
ext_current_temperature: 11
ac_mode: false
current_power: null
current_power_max: null
**saved_preset_mode: boost**
saved_target_temp: 7
saved_hvac_mode: heat
motion_sensor_entity_id: null
motion_state: null
power_sensor_entity_id: null
max_power_sensor_entity_id: null
overpowering_state: null
presence_sensor_entity_id: binary_sensor.presence_domicile
presence_state: off
window_state: on
window_auto_state: off
window_bypass_state: false
window_sensor_entity_id: binary_sensor.fenetre_chambre_ouverture
window_delay_sec: 30
window_auto_enabled: false
window_auto_open_threshold: 3
window_auto_close_threshold: 0
window_auto_max_duration: 30
window_action: window_turn_off
security_delay_min: 60
security_min_on_percent: 0.5
security_default_on_percent: 0.1
last_temperature_datetime: 2024-11-07T11:30:23.763652+01:00
last_ext_temperature_datetime: 2024-11-07T11:30:23.763652+01:00
security_state: false
minimal_activation_delay_sec: 10
device_power: 1000
**mean_cycle_power: 1000**
total_energy: 256.41
last_update_datetime: 2024-11-07T11:30:23.816118+01:00
timezone: Europe/Paris
temperature_unit: °C
**is_device_active: true**
ema_temp: 18.4
is_used_by_central_boiler: false
temperature_slope: 0
hvac_off_reason: null
is_over_switch: true
is_inversed: false
keep_alive_sec: 0
underlying_entities: switch.fil_pilote_qubino_chambre_fil_pilote_qubino_chambre_switch
**on_percent: 1
power_percent: 100
on_time_sec: 300
off_time_sec: 0**
cycle_min: 5
function: tpi
tpi_coef_int: 0.6
tpi_coef_ext: 0.01
friendly_name: Thermostat Chambre
supported_features: 401

Regards

Yann

**Comment by jmcollin78** [comment]
Hello @yannc74 

What is your release ?

I already have answered this case many time before, can you have a look if your case is not the same. Thank you

**Comment by yannc74** [comment]
Hello, 

My Ha and Versatile are up-to-date (latest releases - updated this morning)

I saw these post afterwards
[accept temperature changes while window is open #273](https://github.com/jmcollin78/versatile_thermostat/discussions/273)
[over valve / window detection #577](https://github.com/jmcollin78/versatile_thermostat/discussions/577)

I need to check some cases but if I summarize behaviors:

1. UseCase 1:
initail state:: window: close , heating: off
Change preset => heating on
open window => heating off
close window => heating on

2. UseCase 2
initail state:: window: open, heating: off
Change preset (by mistake)=> heating on
close window => heating on
open window => on ? I must check.

Yet, I'm not clear how I can control the heating (off) if the window is open and restore the state to heating to "on" when I close the window, by automation. 
It will be a bypass.

For a end-user,  when I configure a window detection, I thinks that the heating will be off if the window is open.
But, with issue 273, it's not the case.
The behavior doesn't seem natural to me. 

So, a question.
Why not control the behavior  [accept temperature changes while window is open #273](https://github.com/jmcollin78/versatile_thermostat/discussions/273) with an option ?
For example:
Configure Vtherm
Window Detection / Option: Allow heating if the window is open: true/false ?

What do you think of my proposal ?
I can try to make this change, if you are agree ?

Thanks

Yann

**Reply by jmcollin78** [reply]
Hello, thank you for your proposal but I would avoid to add more options and complexify the configuration which is already too complex.

The behavior is the following:
1. when you open the window and you have configured to turm off, the last_hvac_mode attribute is saved:
```
saved_hvac_mode: heat
```
2. when the window is closed, the hvac_mode attribute is simply restored.

Now, if you change the preset between 1. and 2. (let says to Boost) the new preset is stored as current preset but hvac_mode stays closed.
When you close the window, hvac_mode is restored to heat and preset is not changed so it stays to boost.

I change a few thing around this to avoid turning on a VTherm when window is open at restart (see https://github.com/jmcollin78/versatile_thermostat/issues/504) but this should have not change the behavior explained above. I will make a test to confirm and I let you know.

**Comment by jmcollin78** [comment, marked as answer]
Hi again,

I just a test and it is working exactmy as described:

1. before opening window (On / Comfort)
<img width="384" alt="Capture d’écran 2024-11-08 à 22 49 42" src="https://github.com/user-attachments/assets/cdcbaf13-ac19-4eb5-b03d-e362fc8af3e4">

2. Open the window (Off / Comfort)
<img width="386" alt="Capture d’écran 2024-11-08 à 22 50 14" src="https://github.com/user-attachments/assets/7f6c4fba-446a-47ab-86b2-5aa0ec88a1a8">

3. Change the preset (Off / Eco)
<img width="385" alt="Capture d’écran 2024-11-08 à 22 50 30" src="https://github.com/user-attachments/assets/f4e96426-a2cb-4468-b358-e9f7774b5c14">

4. Close the window (On / Eco)
<img width="391" alt="Capture d’écran 2024-11-08 à 22 51 49" src="https://github.com/user-attachments/assets/217c1569-8ddd-4d95-8dfd-a3c7395ef2ff">

This is exactly what is expected and this is a good behavior: if preset change during window open, it is taken into account when the window will be close, so that, changes by Scheduler will by applied.

Try it yourself to be convience. So I don't have any explaination of your behavior in the first post. Mayeb your scheduler turn on the VTherm and not only change the preset ? In that case, the Vtherm will be turned on (this is a user choice)

**Comment by yannc74** [comment]
Hello, 

understood !

After many tests with the scheduler, I have the expected behavior.

Thanks !
